### DISCUSSION TITLE
string literal with parentheses (t) is not matched

### DISCUSSION THREAD

**Discussion by mxschumacher** [BurntSushi/ripgrep#1621] [General] [answered]
#### What version of ripgrep are you using?

ripgrep 12.1.1
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)

#### How did you install ripgrep?

with yay (pacman) from AUR

#### What operating system are you using ripgrep on?

Manjaro Linux with the 5.7.0 kernel

#### Describe your bug.

No literal match for pattern with parentheses

#### What are the steps to reproduce the behavior?

Create a text file with the content "row_to_json(t)", call it "test_file"

will match the line:
rg "row_to_json" test_file

will not match the line (note the parentheses)
rg "row_to_json(t)" test_file

#### What is the actual behavior?

```
DEBUG|grep_regex::literal|crates/regex/src/literal.rs:58: literal prefixes detected: Literals { lits: [Complete(row_to_jsont)], limit_size: 250, limit_class: 10 }
DEBUG|globset|crates/globset/src/lib.rs:431: built glob set; 0 literals, 0 basenames, 12 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
```

e.g. no output

#### What is the expected behavior?

It should have matched the line & returned it with a line number.
grep (GNU grep) 3.4 shows the expected behavior

**Comment by okdana** [comment, marked as answer]
>  grep (GNU grep) 3.4 shows the expected behavior

Only because it uses BRE (`grep -G`) by default. `rg`'s regex syntax is similar to PCRE (`grep -P`) or even ERE (`grep -E`)

The syntax is described here: <https://docs.rs/regex/*/regex/#syntax>

Basically you should do either `rg 'row_to_json\(t\)'` or `rg -F 'row_to_json(t)'`
